test/Test/Util/FS.hs

```diff
@@ -228,6 +237,74 @@ approximateEqStream (UnsafeStream infoXs xs) (UnsafeStream infoYs ys) =
       (Finite, Finite)     ->  xs == ys
       (_, _)               -> False
 
+{-------------------------------------------------------------------------------
+  List directory
+-------------------------------------------------------------------------------}
+
+data DirEntry a = Directory a | File a
+  deriving stock (Show, Eq, Ord, Functor)
+
+-- | List all files and directories in the given directory and recusively in all
+-- sub-directories.
+listDirectoryRecursive ::
```

Comment:
Do we need some sort of test to make sure that these listings are actually complete?

Reply:
Preferably, but I have them on a separate branch. The test are quite chunky so I want to add them in a different PR at some point. For now I can only give you my word that they work well enough :stuck_out_tongue:

Reply:
Maybe leave a TODO?